### PR TITLE
feat: add errorFilter option to bypass incrementing failure stats

### PR DESCRIPTION
**Pull request by lance**
Refs: https://github.com/nodeshift/opossum/issues/262

~Still requires documentation.~

### CODE REVIEW

**Comment by lance** [comment]
@RidgeA @ComputerWolf @jahredhope please take a look at the requested feature as implemented and let me know what you think.

File: index.js [hunk 1]
```diff
@@ -51,6 +51,12 @@ const defaults = {
  * can open. This is similar to `options.allowWarmUp` in that no matter how many
  * failures there are, if the number of requests within the statistical window
  * does not exceed this threshold, the circuit will remain closed. Default: 0
+ * @param options.errorFilter {Function} an optional function that will be
+ * called when the circuit 's function fails (returns a rejected Promise). If
```
Comment (jahredhope) [inline index.js]: Is the extra space before each `'s` in this paragraph intentional? 
```suggestion
 * called when the circuit's function fails (returns a rejected Promise). If
```

**Comment by RidgeA** [comment]
@lance 
This is exactly what I expecting. Looking forward to a landing of this PR.

I have only one doubt - wouldn't be better to accept as an `errorHandler` promise/async function as well?
Personally I don't nee this at the moment, but it could be useful in some cases?...

**Comment by ComputerWolf** [comment]
@lance, yes, thank you for taking the time to implement this!

File: index.js (hunk 1 above)
Comment (lance) [reply index.js]: Good catch. My IDE did some strange reformatting when I copied/pasted. Thought I had caught it all.

**Comment by lance** [comment]
@RidgeA I'm concerned about asynchronous execution here - not sure why exactly, but in my thinking, the `errorFilter` function should be simple and straightforward, only checking the result of a received error. I can imagine a scenario where you _might_ want to call a remote service to determine if the error should really be considered an error. But at first blush, that sounds overly complex and might indicate that the architecture should be reconsidered.

**Comment by RidgeA** [comment]
@lance 
Totally agree, this is could be redundant.
Anyway, if someone will need this feature - they could open a new issue and make PR to enhance the `errorHandler`

**Comment by lance** [comment]
@RidgeA @ComputerWolf @jahredhope this has been merged. It will be a few weeks before it lands in a release, but you should be able to use it in a package.json now directly from GitHub.

**Comment by RidgeA** [comment]
@lance 
Thanks a lot!

**Comment by RidgeA** [comment]
Hi @lance,
could you give any forecast when these changes will be in npm?
My company rules prohibit the use of public packages from git repos directly, only NPM...

Thanks!

**Comment by RidgeA** [comment]
@lance, would appreciate, thank you!

**Comment by lance** [comment]
@RidgeA I just published v1.11.0 which includes this change.

**Pull request merged**